src/lib.rs

```diff
@@ -61,6 +63,41 @@ impl Script {
 impl_from_core_type!(Script, BitcoinScriptBuf);
 impl_from_ffi_type!(Script, BitcoinScriptBuf);
 
+#[derive(Debug, Clone)]
+pub struct TxOut {
+    pub value: Arc<Amount>,
+    pub script_pubkey: Arc<Script>,
+}
+
+impl From<BitcoinTxOut> for TxOut {
+    fn from(tx_out: BitcoinTxOut) -> Self {
+        TxOut {
+            value: Arc::new(Amount(tx_out.value)),
+            script_pubkey: Arc::new(Script(tx_out.script_pubkey)),
+        }
+    }
+}
+
+impl From<TxOut> for BitcoinTxOut {
+    fn from(tx_out: TxOut) -> Self {
+        let value = match Arc::try_unwrap(tx_out.value) {
+            Ok(val) => val.0,
+            Err(arc) => arc.0,
```

Comment:
Do you know why the clone is not required here?  I would have thought it was because it's a simple type, but in fact it's an `Amount`, so I'm not clear on why the clone is not required here.

Reply:
> Thanks tnull, this PR has been updated to use your suggestion here!
> 
> _the only small difference is I happened to be able to also use `arc.0` instead of `arc.0.clone()` for `Amount` since it implements the `Copy` trait from what I can see in [bitcoin_units](https://docs.rs/bitcoin-units/0.1.2/src/bitcoin_units/amount.rs.html#856)_

Good q! Wondered this myself when I `cargo clippy`'d and it gave me a warning I could remove the `.clone()` [This comment](https://github.com/bitcoindevkit/bitcoin-ffi/pull/6#discussion_r1750502094) might have been hidden since I "resolved the conversation" is it the sort of thing your looking for (or did you see it and were looking for additional info)?

Reply:
The `bitcoin` `Amount` implements `Copy`, so the `clone` is implicit

Reply:
Oh I see it has to do with the `Copy` trait mentioned in your previous resolved comment (sorry it was hidden from the UI so I didn't see it). Ok cool thanks!

Reply:
no prob, glad you caught that little change while you were reviewing and asked about it anyway to double check me